test/E2ETest/TestHelpers.cs

```diff
@@ -33,5 +43,91 @@ public static string GetSolutionRootDirectory(string solution)
 
             throw new Exception($"Solution file {solution}.sln could not be found in {applicationBasePath} or its parent directories.");
         }
+
+        public static async Task StartHostAndWaitForReplicasToStart(TyeHost host)
```

Comment:
I like the approach used here better than polling with random HTTP clients and retries 👍